backend/src/database/state-migrations/8-treatment-system-improvements.ts

```diff
@@ -0,0 +1,118 @@
+import type {
+    ExerciseState,
+    ImageProperties,
+    Mutable,
+} from 'digital-fuesim-manv-shared';
+import {
+    cloneDeepMutable,
+    SpatialTree,
+    StrictObject,
+} from 'digital-fuesim-manv-shared';
+import type { Migration } from './migrations';
+
+export const treatmentSystemImprovements8: Migration = {
```

Comment:
I don't really like that you're exclusively using `any` here.

Reply:
I believe `any` is the best solution here. We don't have typings for the previous state. Adding the typings manually like you did for the previous migrations is not a very nice way (more code, more complexity, creating the typings is also error prone). In addition, you will constantly run into problems because you need to e.g rename properties etc.
Migrations won't (or shouldn't) break if you change other parts of the codebase. Therefore one of typescripts biggest features (automatic refactoring) isn't of any use here.

TLDR: the cost/benefit for completely typed out code is not very good for the migrations.

Reply:
Yes, true. I agree that the whole system is not very good. Also that you get the initial state for action migrations is worthless as this is the state after all migrations, which could be vastly different than the state you expected when writing the migration. At some point I'll refactor this a bit, I think.

Reply:
> Yes, true. I agree that the whole system is not very good. Also that you get the initial state for action migrations is worthless as this is the state after all migrations, which could be vastly different than the state you expected when writing the migration. At some point I'll refactor this a bit, I think.

Shouldn't you get the state with the same version as the actions (so in this case 7?) instead of "the state after all migrations" (a state version >= 7)?
Or do I misunderstand your comment?

Reply:
This is the behavior you'd expect, correct. But this is not the current behavior. Currently, you get the state after the last migration.

Reply:
> This is the behavior you'd expect, correct. But this is not the current behavior. Currently, you get the state after the last migration.

That should be changed. This not only makes it unintuitive but could also introduce bugs if the state is changed and makes the code much harder to maintain.

Is there already an issue for this?

Reply:
I've added #524.